pkg/agent/params.go

```diff
@@ -208,6 +213,11 @@ func assignKubernetesParameters(properties *datamodel.Properties, parametersMap
 			addValue(parametersMap, "networkPlugin", kubernetesConfig.NetworkPlugin)
 			addValue(parametersMap, "networkMode", kubernetesConfig.NetworkMode)
 			addValue(parametersMap, "containerRuntime", kubernetesConfig.ContainerRuntime)
+			if kubernetesConfig.ContainerRuntime == "containerd" {
+				addValue(parametersMap, "cliTool", "ctr")
+			} else {
+				addValue(parametersMap, "cliTool", "docker")
+			}
```

Comment:
It's a bit confusing to see two places assigning the same parameters. Can we just do it from agent pool profile? I kind of imagine the agent pool profile should be updated in RP side already, if it needs to inherit any property from cluster.

Not a blocking issue but worth consideration , along with the teleport enablement flag as we discussed in the design review - historically container runtime is a cluster level setting as we don't have multi agent pool, but now, most of the setting becomes agent pool specific, e.g os config, container run time, k8s version, etc. Long term, I think it's a better design to completely move them from cluster to agent pool. The cluster level setting is just good for control plane side components. WDYT?

Reply:
Agreed. Will remove the addition here. We should let aks-rp handle all the agentpool profile configurations. 

We can probably also remove the line above where it sets the runtime but I will leave it in for now, to be safe. At GA aks-rp will set the appropriate agentpool-level runtime configuration based on the VHD distro so this should def just be an agentpool-level config here as well.

I'll make an ADO item to track this small cleanup task post-GA.